### DISCUSSION TITLE
Added: LDM Notes for November 1st, 2021

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#5375] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2021/LDM-2021-11-01.md

## Agenda

1. Order of evaluation for Index and Range
2. Collection literals

**Comment by HaloFour** [comment]
Java added similar helper methods which create immutable implementations of the common collections.

```java
List<String> list = List.of("foo", "bar", "baz");
Set<String> set = Set.of("foo", "bar", "baz");
Map<Integer, String> map = Map.of(1, "foo", 2, "bar", 3, "baz");
```

**Reply by Joe4evr** [reply]
Looks a lot like the `Immutable[(Array|List|HashSet|etc)].Create` methods.

**Comment by Joe4evr** [comment]
> After a bit of discussion, we unanimously agreed to standardize the order of evaluation to be
consistent with the fully written out form of `expr.End.GetOffset(receiver.Length) - start`: ie `expr`, then `receiver`, then `Length`.
> 
> #### Conclusion
> 
> Codify the expected behavior as `receiver` then `expr` then `Length`.

Uhhh..... There are still 2 different orderings in this excerpt. 🙃

**Reply by 333fred** [reply]
The conclusion is correct.

**Reply by 333fred** [reply]
The notes have been updated.

**Comment by KarolKalbarczyk** [comment]
Kotlin actually plans on adding collection literals (with very similar semantics to what is in the current C# proposal).
https://youtrack.jetbrains.com/issue/KT-43871

**Reply by jnm2** [reply]
Ooh, nice. They also have the syntax I'm really hoping for dictionary literals to end up with.

**Comment by huoyaoyuan** [comment]
I want you to utilize static interface methods (or shapes/extensions) to codify the requested members for collections.
